### ISSUE TITLE
Reverser on full powershift vehicles

### ISSUE THREAD

**Issue by KeyboardFarmer** [modelleicher/FS19_realManualTransmission#62]
Hi there, maybe this has been asked/suggested before but is there a reason why the Quadtrac, 8R, 7R, T9 etc don't use the reverser? In real life they all use it, like the John Deere E23 transmission which has left hand shuttle. Even the older John Deeres with full powershfit (Eg. 8400/8410/8520/8530) has a right hand shuttle/reverser.
Is it not possible to implement this with full powershift?

Just a small thought,
KBF

**Comment by Farmboy1539** [comment]
Hey I noticed this the other day too. I'm not sure why they weren't coded for it whether he forgot or just wasn't sure. I was using the 6M and saw the gears and ranges are flipped too. It should have had the gears being 1-4 and the ranges being A-E. A few others had this issue too. 
I went through and fixed the .xml file and tweaked the range ratios a bit to make them more true to life for my personal use. If it's ok with Modelleicher I can send you my fixed xml file. Modelleicher if you want the file I'd be happy send it to you as well.

**Comment by KeyboardFarmer** [comment]
Hi it would be great if you shared it with Modelleicher, so it's updated in the basegameConfigs.xml

Regards,
KBF

**Comment by modelleicher** [comment]
Most of the transmissions in the basegame.xml were user-made so I can't tell you why they are the way they are. But if you send me the fixed file I can include it in the next RMT version :)

**Comment by Einfllslos** [comment]
Hi
I think the main problem is still not solved.
If you would use a reverser with RMT in a fullpowershift vehicle with more forward then reverse gears, you get all gears even the reverse ones forward and backward. 
Without reverser you have to shift through all forward gears to get in reverse and I think thats annoying.

So jetzt nochmal auf Deutsch
In echt haben Volllastschaltgetriebe ja auch eine Wendeschaltung. Man legt den Hebel um und der Traktor fährt automatisch rückwärts. (Meistens kann man programmieren in welchem Rückwärtsgang der Traktor anfahren soll)
Wenn ich jetzt eine Wendeschaltung in RMT einbaue, gleichzeitig aber Rückwärts- und Vorwärtsgänge definiert habe, kann ich sowohl die Rückwärts als auch die Vorwärtsgänge in beide Richtungen fahren. Der reverser unterscheidet da also nicht zwischen Rückwärtsgang (isReverse="true") und Vorwärtsgang.
Ohne reverser muss ich dann durch alle Gänge "rühren" um vom sagen wir mal 10 Gang (vorwärts) in den dritten Rückwärtsgang zu kommen.

**Comment by Farmboy1539** [comment]
Hey guys I'm sorry I haven't had a chance to send you the file yet, we've been calving like crazy these last two weeks. I've just been trying to tidy it up a bit before I send it on to you especially since you want to include it. I should have it to you by the end of the weekend for sure. 

Einfllslos, So at first I didn't really understand what you were saying but I think I've got it now, especially after I google translated your German. : )  I didn't catch this until you brought it up and you're right using isReverse=true should tell it it's a reverse gear only I assume. I tried putting in isReverse=false for all the forward only gears and that still didn't make a difference allowing the tractor to move in reverse through all the forward gears. Maybe Modelleicher can shed some light on this or maybe that part isn't fully working just yet? 

Of course this only applies to the new electronic power shift transmissions. The old power shift transmissions on the John Deere Iron Horse Series (4440's and such) you had to physically come down through all the forward gears to get it into reverse.

**Comment by modelleicher** [comment]
@Einfllslos 
Der Reverser so wie er in RMT implementiert ist ist eine klassische Wendeschaltung, d.h. man hat ein Wendegetriebe evtl. mit einer Übersetzung und so werden alle Gänge in vorwärts und rückwärts unterteilt.
Das was du beschreibst (programmieren in welchem Gang angefahren werden soll) ist ja so zumindest ingame nicht möglich.
D.h. an sich müsste ich einen Reverser-Modus verbauen in dem man einfach angeben kann in welche Gang beim vorwärts und welchen Gang beim rückwärts schalten geschaltet werden soll?
Oder speichert ein solcher Reverser auch in echt in welchem Gang man zuletzt war und legt den dann wieder ein?


@Farmboy1539 
Since RMT was initially only meant to be a manual gearbox all the XML attributes and "logic" behind those work like a classic gearbox.
So if you make a gear or range into reverse by isReverse="true" than this gear or range is a reverse gear/range which means if you are in that gear or range you're going reverse.
It isn't meant as a "this can't be used in reverse - this can be used in reverse" type of attribute.

The reverser is a classic reverser which is usually put in front (or at the end) of the main transmission and thus can shift the entire transmission into reverse no matter what gear (IRL) for most classic and even modern transmissions, partial powershift, non-powershift this is how it works.
Thats the basis of the implementation for FS.

I think the FPS transmissions are the odd ones out here and I didn't really account for those in the first place.
As you put it, IRL with old FPS transmissions you have to actually physically shift through all the ranges to go into reverse, thats exactly the way it works in RMT right now.

So in short, RMT isn't really working for most of the stuff after the mid 80s or so.. lol :D 


That said, a modern FPS transmission has all kind of automatic shifting and stuff none of which RMT can do right now.
